### PR TITLE
feat: Add `n_bonacci.cpp` in math section

### PR DESCRIPTION
**Pull request by Swastyy**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Fixes for https://github.com/TheAlgorithms/C-Plus-Plus/projects/6#card-63413779

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1544"></a>

### CODE REVIEW

File: math/n_bonacci.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * @details
+ * In general, in N-bonacci sequence,
+ * we generate sum of preceding N numbers from the next term.
+ *
+ * For example, a 3-bonacci sequence is the following:
+ * 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ * In this code we take N and M as input where M is the number of terms
+ * to be printed of the N-bonacci series
+ *
+ * @author [Swastika Gupta](https://github.com/swastyy)
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:14]:
```suggestion
 * @author [Swastika Gupta](https://github.com/Swastyy)
```

File: math/n_bonacci.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ */
+namespace math {
+/**
+ * @namespace n_bonacci
+ * @brief Functions for the [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers)
+ * implementation
+ */
+namespace n_bonacci {
+/**
+ * @brief Finds the N-Bonacci series for the `n` parameter value and `m` parameter terms.
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:33]:
```suggestion
 * @brief Finds the N-Bonacci series for the `n` parameter value and `m` parameter terms
```

File: math/n_bonacci.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ * @param n is in the N-Bonacci series
+ * @param m is the number of terms in the N-Bonacci sequence
+ * @returns the n-bonacci sequence as vector array
+ */
+std::vector<int> N_bonacci(int n, unsigned int m) {
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:38]: Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

File: math/n_bonacci.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();  // run self-test implementations
+    return 0;
+}
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:98]:
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Good work! 👍
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/Swastyy/C-Plus-Plus/actions

File: math/n_bonacci.cpp [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+ * In general, in N-bonacci sequence,
+ * we generate sum of preceding N numbers from the next term.
+ *
+ * For example, a 3-bonacci sequence is the following:
+ * 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ * In this code we take N and M as input where M is the number of terms
+ * to be printed of the N-bonacci series
+ *
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:16]:
```suggestion
 */

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
LGTM. Great work, @Swastyy; thank you! 😄👍🎉

**Comment by Swastyy** [comment]
@mishraabhinn See this PR has only my file. Check this out.

**Review by Panquesito7 — COMMENTED** [review]
If nobody has any objections, I'll merge this PR later today.

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**